wireddown/qt-py-s3-daq-app
# Initialize package, configure poetry, configure tools, add tests

## Summary

This change introduces the `qtpy_datalogger` package and example acceptance tests.

- Configure `poetry` to create venvs in the project folder
- Add and configure developer tools
  - `pytest`
  - `ruff`
  - `pyright`
 - Configure helper tasks in `poe`
   - `poe lint` to run the analyzers
   - `poe test` to run the tests

## Design

These changes follow the usage and instructions for each tool.

## Screenshots or logs

```pwsh
> qtpy-datalogger    
X:\dev\mux\git\github\qt-py-s3-daq-app\src\qtpy_datalogger\__main__.py
```

## Testing

- Use `importlib` to confirm that Python can import the package
- Use `runpy` to confirm that Python can invoke the package

## Checklist

- [ ] ~~Issues linked / labels applied~~
  - Updated the Contributing wiki page
- [x] Tests added / updated / removed
- [x] Analyzers passed
- [x] Ready to complete
